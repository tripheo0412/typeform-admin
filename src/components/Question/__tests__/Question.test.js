import React from 'react';

import { mount, shallow } from 'enzyme';

import { Question } from '..';

describe('Question', () => {
  let wrapper;
  const setQuestionText = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation(init => [init, setQuestionText]);

  beforeEach(() => {
    wrapper = mount(<Question />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Question should render correctly', () => {
    wrapper = shallow(<Question />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Question should render correctly prop type', () => {
    wrapper = mount(<Question type="email" />);
    expect(wrapper.props().type).toBe('email');
  });
  // it('Question setQuestionText method should be called when text is changed', () => {
  //   const mockedEvent = { preventDefault() {}, target: { value: 'adcb' } };
  //   wrapper
  //     .find('textarea')
  //     .first()
  //     .simulate('change', mockedEvent);
  //   expect(setQuestionText).toHaveBeenCalled();
  // });
});
