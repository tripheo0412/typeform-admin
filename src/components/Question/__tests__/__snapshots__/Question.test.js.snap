// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Question should render correctly 1`] = `ShallowWrapper {}`;
